lib/nfs_v3.c

```diff
@@ -1128,6 +1129,11 @@ nfs3_mount_async(struct nfs_context *nfs, const char *server,
         free(nfs->nfsi->server);
 	nfs->nfsi->server = new_server;
         
+#ifdef HAVE_TLS
+	free(nfs->rpc->server);
+	nfs->rpc->server = strdup(nfs->nfsi->server);
```

Comment:
Check for NULL?

Reply:
sorry, which one?
nfs->nfsi->server is guaranteed to not be NULL because of the above code.
since free(NULL) is a no-op, we are fine with nfs->rpc->server being NULL.
nfs is not NULL, nfs->rpc is also being used already so that also cannot be NULL.

Reply:
Ok, I did not check for that, if it is fine if nfs->rpc->server might be NULL later if strdup fails. Then that is fine.
Thanks for explaining.